### PR TITLE
Feature/71 disable search button

### PR DESCRIPTION
**Pull request by Miguel7373**

### CODE REVIEW

File: src/main/resources/static/js/ButtonDisabler.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+document.addEventListener("DOMContentLoaded", function () {
+    const examNumberInput = document.getElementById("examNumber");
+    const submitButton = document.getElementById("submitButton");
+    function updateButtonState() {
+        const examNumber = examNumberInput.value;
+        submitButton.disabled = examNumber.length <= 4 || examNumber === "";
```
Comment (RandomTannenbaum) [inline src/main/resources/static/js/ButtonDisabler.js:6]: Only check for empty

**Review by RandomTannenbaum — CHANGES_REQUESTED** [review]

File: src/main/resources/static/js/ButtonDisabler.js (hunk 1 above)
Comment (Miguel7373) [reply src/main/resources/static/js/ButtonDisabler.js]: Ive have changed it to only check for empty and made the test so they test exactly that

File: src/main/resources/static/js/ButtonDisabler.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+document.addEventListener("DOMContentLoaded", function () {
+    const examNumberInput = document.getElementById("examNumber");
+    const submitButton = document.getElementById("submitButton");
+    function updateButtonState() {
+        const examNumber = examNumberInput.value;
+        submitButton.disabled = examNumber === "";
```
Comment (RandomTannenbaum) [inline src/main/resources/static/js/ButtonDisabler.js:6]: Use `.trim()` to account for whitespaces

File: src/test/e2e/cypress/e2e/search.cy.js [hunk 3]
```diff
@@ -81,3 +81,12 @@ it('should rename files to subject-folder they are inside of after downloading',
     }
     cy.expect(i).to.equal(3)
 });
+
+it('should disable button if not enough numbers are in input', () => {
```
Comment (RandomTannenbaum) [inline src/test/e2e/cypress/e2e/search.cy.js:85]: Change test name

File: src/test/e2e/cypress/e2e/search.cy.js [hunk 4]
```diff
@@ -81,3 +81,12 @@ it('should rename files to subject-folder they are inside of after downloading',
...
+    cy.get('input[type="text"]').type('110');
+    cy.get('button').should('not.be.disabled')
+});
+
+it('should disable button if input is empty', () => {
+    cy.get('button').should('be.disabled')
```
Comment (RandomTannenbaum) [inline src/test/e2e/cypress/e2e/search.cy.js:91]: Maybe extend test to test with whitespaces once `.trim()` is added

**Review by RandomTannenbaum — CHANGES_REQUESTED** [review]

File: src/test/e2e/cypress/e2e/search.cy.js (hunk 4 above)
Comment (Miguel7373) [reply src/test/e2e/cypress/e2e/search.cy.js:93]: added white space to test

File: src/main/resources/static/js/ButtonDisabler.js (hunk 2 above)
Comment (Miguel7373) [reply src/main/resources/static/js/ButtonDisabler.js]: added .trim

File: src/test/e2e/cypress/e2e/search.cy.js (hunk 3 above)
Comment (Miguel7373) [reply src/test/e2e/cypress/e2e/search.cy.js]: done

**Review by RandomTannenbaum — APPROVED** [review]

**Review by pizzi-cato — DISMISSED** [review]

**Review by pizzi-cato — APPROVED** [review]

**Pull request merged**